deployment/asset/artem.1
.ie \n(.g .ds Aq \(aq
.el .ds Aq '
.TH artem 1  "artem 0.6.0" 
.SH NAME
artem \- artem is a small cli program written in rust to easily convert images to ascii art.
.SH SYNOPSIS
\fBartem\fR [\fB\-\-help\fR] [\fB\-V\fR|\fB\-\-version\fR] [\fB\-c\fR|\fB\-\-characters\fR] [\fB\-s\fR|\fB\-\-size\fR] [\fB\-h\fR|\fB\-\-height\fR] [\fB\-w\fR|\fB\-\-width\fR] [\fB\-\-ratio\fR] [\fB\-\-flipX\fR] [\fB\-\-flipY\fR] [\fB\-o\fR|\fB\-\-output\fR] [\fB\-\-thread\fR] [\fB\-\-invert\fR] [\fB\-\-background\fR] [\fB\-\-border\fR] [\fB\-\-no\-color\fR] [\fB\-\-verbose\fR] <\fIINPUT\fR> 
.SH DESCRIPTION
artem is a small cli program written in rust to easily convert images to ascii art.
.SH OPTIONS
.TP
\fB\-\-help\fR
Print help information
.TP
\fB\-V\fR, \fB\-\-version\fR
Print version information
.TP
\fB\-c\fR, \fB\-\-characters\fR
Change the characters that are used to display the image.The first character should have the highest \*(Aqdensity\*(Aq and the last should have the least (probably a space).A lower detail map is recommend for smaller images.
.TP
\fB\-s\fR, \fB\-\-size\fR [default: 80]
Change the size of the output image.The minimum size is 20, the maximum 230. Values outside of the range will beignored and changed to the nearest usable value. This argument is conflicting with \-\-width and \-\-height
.TP
\fB\-h\fR, \fB\-\-height\fR
Use the terminal maximum terminal height to display the image.This argument is conflicting with \-\-size and \-\-width 
.TP
\fB\-w\fR, \fB\-\-width\fR
Use the terminal maximum terminal height to display the image.This argument is conflicting with \-\-size and \-\-height 
.TP
\fB\-\-ratio\fR [default: 0.42]
Change the ratio between height and width, since Ascii chars are a bit higher than long.The default value is 0.43, min is 0 and max 2. It is not recommend to change this setting.
.TP
\fB\-\-flipX\fR
Flip the image along the X axis
.TP
\fB\-\-flipY\fR
Flip the image along the Y axis
.TP
\fB\-o\fR, \fB\-\-output\fR
Output file for non\-colored ascii. If the output file is a plaintext file, no color will be used. The use color, either use a file with an .ansi extension, or an .html file, to convert the output to html.
.TP
\fB\-\-thread\fR [default: 4]
OutputNumber of threads used to convert the image. A larger number can lead to grater performance. Defaults to 4
.TP
\fB\-\-invert\fR
Inverts the characters used for the image, so light characters will as dark ones. Can be useful if the image has a dark background.
.TP
\fB\-\-background\fR
Sets the background of the ascii as the color. This will be ignored if the terminal does not support truecolor. This argument is mutually exclusive with the no\-color argument.
.TP
\fB\-\-border\fR
Adds a decorative border surrounding the ascii image. This will make the image overall a bit smaller, since it respects the user given size.
.TP
\fB\-\-no\-color\fR
Do not use color when printing the image to the terminal.
.TP
\fB\-\-verbose\fR
Choose the verbosity of the logging level.
.TP
Path to the target image. Does NOT alter the original image
.SH VERSION
v0.6.0


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

# [Unreleased]
### Added
- A [homebrew tap](https://github.com/FineFindus/homebrew-tap) is now available
- Cargo publishing is now  done in the release workflow
- The release workflow now updates the homebrew tap

### Changed
- Overhauled the installation section in the README, it now contains much more detailed installations instructions
## [0.6.1] - 2022-03-24

### Added

- Linux Binaries will now also be compiled with `musl`
- Completion scripts for the `.deb``will be copied in the CD process
- With mscv compiled windows binaries are available as an alternative to the gnu compiled ones
- MacOS binaries for (x86 and arm) have been added to the CD process

## [0.6.0] - 2022-03-24

### Added

- When using an html output file, artem will now converted the result to html, this also works with .ans files respectively
- More Documentation to better describe the code
- The `--border` flag can be used to create a border around the ascii image
- The `--flipX` flag can be used horizontally flip the image
- The `--flipY` flag can be used vertically flip the image
- Two more tests, which fully compare the results

### Changed

- Major refactoring

## [0.5.1] - 2022-03-14

### Changed

- Using a new workflow job for the windows build

## [0.5.0] - 2022-03-14

### Added

- Release builds are now available for more targets (linux x64 and arm) and windows (using gnu-target)

### Changed

- Using the `--width` argument now correctly resizes the image
- Using the `--height` argument now uses the correct height of the terminal
- Using multiple Threads now display the full image instead of leaving a few pixels out
- Updated the example image in the README to reflect the changes

## [0.4.1] - 2022-03-01

### Added

- Changed version to 0.4.1, since github actions would not use the right files otherwise
- Fixed error with tar command in cd

## [0.4.0] - 2022-03-01

### Added

- README now contains an installation section
- Use the `--background` flag to let the ascii chars have a background color. Might be useful for images with dark backgrounds.
- Use the `--invert` flag to change the brightness of the used characters. Can be useful for images with a dark background
- README now lists some example formats that can be used
- Tab completions now works in other shells as well (fish and zsh in deb package)
- Removed linting problems found by clippy
- CI tests now against the stable, beta and nightly rust version
- CI now checks for clippy warnings
- Changelog file to document changes to the project
- A Feature template can be used to easily request features over Github

### Changed

- Logging no longer logs the date, since it is not needed
- Man Page String are now formatted correctly

## [0.3.0] - 2022-02-25

### Added

- Logging with different verbosity levels to help debugging
- `verbose` flag can be used to change the verbosity, defaults to `error`


deployment/asset/artem.bash
_artem() {
    local i cur prev opts cmds
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"
    cmd=""
    opts=""

    for i in ${COMP_WORDS[@]}
    do
        case "${i}" in
            "$1")
                cmd="artem"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        artem)
            opts="-V -c -s -h -w -o --help --version --characters --size --height --width --ratio --flipX --flipY --output --thread --invert --background --border --no-color --verbose <INPUT>"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --characters)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -c)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --size)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -s)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --ratio)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --output)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -o)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --thread)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --verbose)
                    COMPREPLY=($(compgen -W "" -- "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

complete -F _artem -o bashdefault -o default artem
